packages/labs/analyzer/src/lib/javascript/classes.ts

```diff
@@ -175,7 +192,7 @@ export const getClassMembers = (
 
       (info.static ? staticFieldMap : fieldMap).set(
         node.name.getText(),
-        new ClassField({
+        new CustomElementField({
```

Comment:
is this right? all properties of all classes are now "custom element" fields?

or am i missing a condition somewhere that means this only happens for lit properties?

Reply:
this chain of logic is what has been itching my mind.

i think what we should be doing (somewhere) is the same as what we do for classes:

```ts
  if (isLitElementSubclass(declaration, analyzer)) {
    return getLitElementDeclaration(declaration, analyzer);
  }
  if (isCustomElementSubclass(declaration, analyzer)) {
    return getCustomElementDeclaration(declaration, analyzer);
  }
```

similar, somewhere around here, we'd do:

```ts
if (isCustomElementSubclass(node, analyzer)) {
  return getCustomElementField(node, analyzer);
}
```

basically, right now you're computing all a class' members. you then _mutate_ those members later on in lit-specific code. which is the bit that irks me.

i feel like you should be computing the right thing the first time around, somehow.

everywhere else, we seem to have achieved that by type-guards and delegations to the associated `get*` code.

Reply:
good catch, there should be a predicate here

Reply:
actually this should always be a `ClassField`, because custom elements and lit elements are handled elsewhere

Reply:
> basically, right now you're computing all a class' members. you then mutate those members later on in lit-specific code. which is the bit that irks me.

i guess the trade-off would be to have lit-specific code in the general `classes` file? we don't have code for attr/prop pairs in `extends HTMLElement` yet

Reply:
i opted to new up, like your snippet - seemed the most expedient. At the same time I broke up `getClassMembers` - to ease future refactoring